core/server/services/auth/oauth.js

```diff
@@ -179,7 +179,13 @@ oauth = {
             accessToken: authUtils.getBearerAutorizationToken(req)
         };
 
-        return oauthServer.token()(req, res, next);
+        return oauthServer.token()(req, res, function (err) {
+            if (err && err.status === 400) {
+                err = new common.errors.ValidationError({err: err, message: err.message});
```

Comment:
It's probably better to return a BadRequestError - if a param is missing, it's syntactically wrong.

Reply:
Updated